util/src/main/java/io/camunda/zeebe/util/health/CriticalComponentsHealthMonitor.java

```diff
@@ -105,26 +105,15 @@ private void calculateHealth() {
     final var previousReport = healthReport;
     healthReport = calculateStatus();
 
-    if (previousReport == healthReport) {
+    if (previousReport.getStatus() == healthReport.getStatus()) {
```

Comment:
:question: I could imagine that sometimes we are setting the status to `UNHEALTHY` first and then adding details such as an underlying exception later. In that case we wouldn't log the details since the status hasn't changed. So maybe we need to check status and issue for equality here?

Reply:
You mean when it becomes unhealthy two times in a row for two different reasons?

Reply:
No, but that's another example! I meant something where a component reports `UNHEALTHY` without setting the issue at first. When it later adds an issue and still reports `UNHEALTHY`, we'd want to log that.

Reply:
Would that happen? 
Anyway, I will change the check to compare two healthReports.